cmd/ship_receiver_plugin.cpp

```diff
@@ -267,11 +292,23 @@ class ship_receiver_plugin_impl : std::enable_shared_from_this<ship_receiver_plu
          auto start_from = utils::to_block_num(head_header->mix_hash.bytes) + 1;
          SILK_INFO << "Canonical header start from block: " << start_from;
 
-         if( start_from_block_id ) {
-            uint32_t block_num = utils::to_block_num(*start_from_block_id);
-            SILK_INFO << "Using specified start block number:" << block_num;
-            start_from = block_num;
+         if (restart) {
+            // Always start from early blocks to handle potential forks.
+            // Clearly this approach will introduce overhaad, but since this piece of code 
+            // will only execute during recovery, it should be fine.
+            if (head_header->number > 250) {
+               start_from -= 500;
```

Comment:
It is trivial to keep track of LIB which would be better than just guessing at a block number. See `block.last_irreversible.block_num`.

Reply:
That means we need a set of functions to query single block etc. 

And there's a corner case:
FORK ------ A (origin HEAD)
| -LIB--------A' (new block at same height) 

In the extreme case that LIB already pass the fork block when we try to restart, LIB is not enough to recover.